src/components/chart components/BarChart.jsx

```diff
@@ -0,0 +1,40 @@
+import React, { useState, useEffect } from 'react';
+import { Chart as ChartJS, BarElement, CategoryScale, LinearScale, Title } from "chart.js";
+import { Bar } from "react-chartjs-2";
+import { generateLabels, generateChartData } from "./chartsFunctions";
+import { generateChartStyle, generateOptions } from "./chartProperties";
+
+ChartJS.register(
+    CategoryScale,
+    LinearScale,
+    BarElement,
+    Title,
+);
+
+export default function BarChart({ timeInterval, chartTitle }) {
+    const [chart, setChart] = useState([]);
+    const url = "http://localhost:3004/transactions";
+
+    useEffect(() => {
+        fetchTransactions();
+    }, []);
+
+    const fetchTransactions = async () => {
+        try {
+            const response = await fetch(url);
+            const res = await response.json();
+            setChart(res);
+        } catch (err) {
+            console.log(err);
```

Comment:
here the fetchTransactions should be used with the same fetch api previously created for users and transactions. and pass in another callback, so you should use that for everything going forward. but besides this i think you should create a container component and a presentational one. the container one should contain the information for the chart, the transactions, url, etc, and the presentational one will contain the generic Line to provide to it all the chart data, not knowing anything about transactions